test/jdk/javax/swing/JSpinner/4515999/JSpinnerMouseAndKeyPressTest.java

```diff
@@ -0,0 +1,169 @@
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JSpinner;
+import javax.swing.SpinnerDateModel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+
+import static javax.swing.UIManager.getInstalledLookAndFeels;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4515999
+ * @summary Check whether incrementing dates via the keyboard (up/down) gives
+ * the same results as using mouse press on the arrow buttons in a JSpinner.
+ * @run main JSpinnerMouseAndKeyPressTest
+ */
+public class JSpinnerMouseAndKeyPressTest {
+    //2 days in milliseconds
+    private static final int EXPECTED_VALUE_2_DAYS = 2 * 24 * 60 * 60 * 1000;
+    private static JFrame frame;
+    private static JSpinner spinner;
+
+    public static void main(String[] s) throws Exception {
+        runTest();
+    }
+
+    private static void setLookAndFeel(final String laf) {
+        try {
+            UIManager.setLookAndFeel(laf);
+            System.out.println("LookAndFeel: " + laf);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void createUI() {
+        frame = new JFrame();
+        JPanel panel = new JPanel();
+        spinner = new JSpinner();
+        spinner.setModel(new DateModel());
+        JSpinner.DateEditor editor = new JSpinner.DateEditor(spinner, "dd/MM/yy");
+        spinner.setEditor(editor);
+        spinner.setComponentOrientation(ComponentOrientation.LEFT_TO_RIGHT);
+        panel.add(spinner);
+        frame.add(panel);
+        frame.setUndecorated(true);
+        frame.pack();
+        frame.setAlwaysOnTop(true);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+
+    public static void runTest() throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoWaitForIdle(true);
+        robot.setAutoDelay(100);
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(UIManager.LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            SwingUtilities.invokeAndWait(() -> {
+                setLookAndFeel(laf);
+                createUI();
+            });
+            try {
+                final Point spinnerLocationOnScreen = spinner.getLocationOnScreen();
+                final int spinnerEditorWidth = spinner.getEditor().getWidth();
+                final int spinnerButtonWidth = spinner.getWidth() -
+                        spinnerEditorWidth;
+                final int spinnerQuarterHeight = spinner.getHeight() / 4;
+
+                Point spinnerUpButtonCenter = new Point();
+                spinnerUpButtonCenter.x = spinnerLocationOnScreen.x + spinnerEditorWidth +
+                        (spinnerButtonWidth / 2);
+                spinnerUpButtonCenter.y = spinnerLocationOnScreen.y +
+                        spinnerQuarterHeight;
```

Comment:
The [Java Style Guidelines](http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html#toc-wrapping-lines) suggests breaking before the operators.